### ISSUE TITLE
Package on Ubuntu 19.10

### ISSUE THREAD

**Issue by nlkduy** [phpmyadmin/phpmyadmin#15515]
Hi guys,

I installed Ubuntu 19.10, but I didn't find phpmyadmin on this link: https://launchpad.net/ubuntu/eoan/+source/phpmyadmin.

Thank you for seeing!

**Comment by williamdes** [comment]
Hi @nlkduy have a look to #15236, you will find a link to our PPA here (https://github.com/phpmyadmin/phpmyadmin/issues/15236#issuecomment-502392098) 
We are working on packaging for Debian and Ubuntu will update after Debian

Am I correct @Blaimi?

**Comment by Blaimi** [comment]
Yep.

I'm not sure if the ppa is working currently for eoan because it doesn't have php-symfony-polyfill.

I'll update a new package without the dependency in the next days along with an update to 4.9.1

**Comment by anburocky3** [comment]
Tried the **PPA** `sudo add-apt-repository ppa:phpmyadmin/ppa`, didn't work for **Ubuntu 19.10**

**Comment by anburocky3** [comment]
Do we have any timeline deliveries for this package?

**Comment by williamdes** [comment]
@Blaimi Can you please publish the package for the disco series on the PPA ?

**Comment by williamdes** [comment]
"since Eoan is past feature freeze and Debian import freeze, phpmyadmin won't be included in the 19.10 release."
From: https://bugs.launchpad.net/ubuntu/+source/phpmyadmin/+bug/1837775

But we will update the PPA to offer support for Ubuntu 19.10

**Comment by Blaimi** [comment]
I'll do my best to find the time on the coming weekend to update the ppa to 4.9.1 for eoan as well as bionic and inbetweeners.

(If I don't, feel free to disturb 😋)

**Comment by anburocky3** [comment]
Waiting for eoan release. (Y)

**Comment by Blaimi** [comment]
the ppa ist up-to-date now :) tested on eoan, bionic and debian buster.

**Comment by nlkduy** [comment]
![Screenshot from 2019-11-02 10-01-54](https://user-images.githubusercontent.com/16206786/68065173-e35cc280-fd57-11e9-9708-48a07f24b23e.png)
Its work! Thank you very much!

**Comment by anburocky3** [comment]
Here is the PPA rep link: 

>**sudo add-apt-repository ppa:phpmyadmin/ppa**

**Comment by OGJAYYOU** [comment]
how do i open the phpmyadmin dashboard im pretty new to ubuntu

**Comment by williamdes** [comment]
@OGJAYYOU if I remember well you need to browse http://localhost/phpmyadmin
But the apache config can help you to find out :)

**Comment by OGJAYYOU** [comment]
wait my phpmyadmin might not be working

**Comment by OGJAYYOU** [comment]
does that sudo add-apt-repository ppa:phpmyadmin/ppa work ??? this link here did not install phpmyadmin

**Comment by williamdes** [comment]
@OGJAYYOU you need to do
```bash
sudo add-apt-repository ppa:phpmyadmin/ppa
sudo apt-get update
sudo apt-get install phpmyadmin
```

**Comment by OGJAYYOU** [comment]
@williamdes thanks boss you did a good job

**Comment by OGJAYYOU** [comment]
my dpkg isnt working anymore need help unistalling phpmyadmin

**Comment by williamdes** [comment]
@OGJAYYOU `apt-get purge phpmyadmin`
but can you explain the issue ?

**Comment by OGJAYYOU** [comment]
ok the thing is i first installed mysql but that was days ago then yesterday when i installed phpmyadmin, the installation process failed with some error and assumed it was because of mysql configuration. now when i managed to uninstall both phpmyadmin and mysql, it didnt remove the databases because when i reinstall the LAMP stack it brings back the old databases plus its not bringing up the terminal configuration window.

**Comment by williamdes** [comment]
@OGJAYYOU are you affected by https://bugs.launchpad.net/ubuntu/+source/phpmyadmin/+bug/1652359?

**Comment by Blaimi** [comment]
> ok the thing is i first installed mysql but that was days ago then yesterday when i installed phpmyadmin, the installation process failed with some error and assumed it was because of mysql configuration. now when i managed to uninstall both phpmyadmin and mysql, it didnt remove the databases because when i reinstall the LAMP stack it brings back the old databases plus its not bringing up the terminal configuration window.

You need to purge mysql also. uninstall is not enough to delete all databases.

**Comment by OGJAYYOU** [comment]
i managed to install phpmyadmin and LAMP stack. however my new error reads

mysqli_real_connect(): (HY000/2054): The server requested authentication method unknown to the client

**Comment by OGJAYYOU** [comment]
i think its because i wrote my system using php and mysqli, so my question is what should i do to perharps not get this error????

**Comment by williamdes** [comment]
@OGJAYYOU seems like #15217  to me

**Comment by williamdes** [comment]
[edit] #14220

**Comment by OGJAYYOU** [comment]
thanks @williamdes i finally made it even though it took me a very long time but i managed to attain what i wished for, you really did a good job brother. i appreciate it

**Comment by OGJAYYOU** [comment]
![21](https://user-images.githubusercontent.com/52784179/68835321-6e2aaf00-06c0-11ea-9420-92e4c13ff204.png)

**Comment by williamdes** [comment]
Ubuntu "focal" 20.04 has phpMyAdmin 4.9.2 !

https://launchpad.net/ubuntu/focal/+package/phpmyadmin

**Comment by williamdes** [comment]
```
[...] we do not reintroduce .deb packages in stable releases that
have been removed in the development cycle.
```
From Steve Langasek who removed the package, ref: https://bugs.launchpad.net/ubuntu/+source/phpmyadmin/+bug/1837775/comments/5

TLDR; use our PPA for 19.10

**Comment by baqwas** [comment]
Hello,

Sorry to comment on a closed topic but I'm a total newbie on many things including **phpMyAdmin**. I did follow the suggestions on appending ppa to my installation (Ubuntu **Eoan 19.10** with all standard _update_/_upgrade_ operations applied currently). I was able to install **phpMyAdmin** successfully with elevated privileges but when I launch a session with the following command:

`http://localhost/phymyadmin/`

from the installed desktop, the following is displayed (only a few lines because it will become apparent, I presume, to any **Apache**/**phpMyAdmin** expert):

```
<?php
/* vim: set expandtab sw=4 ts=4 sts=4: */
/**
 * Main loader script
 *
 * @package PhpMyAdmin
 */
use PhpMyAdmin\Charsets;
use PhpMyAdmin\Config;
use PhpMyAdmin\Core;
use PhpMyAdmin\Display\GitRevision;
use PhpMyAdmin\LanguageManager;
use PhpMyAdmin\Message;
use PhpMyAdmin\RecentFavoriteTable;
use PhpMyAdmin\Relation;
use PhpMyAdmin\Response;
use PhpMyAdmin\Sanitize;
use PhpMyAdmin\Server\Select;
use PhpMyAdmin\ThemeManager;
use PhpMyAdmin\Url;
use PhpMyAdmin\Util;
use PhpMyAdmin\UserPreferences;

/**
 * Gets some core libraries and displays a top message if required
 */
require_once 'libraries/common.inc.php';

/**
 * pass variables to child pages
 */
$drops = array(
    'lang',
    'server',
    'collation_connection',
    'db',
    'table'
);
foreach ($drops as $each_drop) {
    if (array_key_exists($each_drop, $_GET)) {
        unset($_GET[$each_drop]);
    }
}
unset($drops, $each_drop);

/*
 * Black list of all scripts to which front-end must submit data.
 * Such scripts must not be loaded on home page.
 *
 */
$target_blacklist = array (
    'import.php', 'export.php'
);

```
All suggestions on remediation or workaround would be appreciated. I feel that my limited knowledge is the handicap here.

Kind regards.

**Comment by williamdes** [comment]
@baqwas you should maybe try to run `sudo apt-get install libapache2-mod-php`

**Comment by baqwas** [comment]
Hello @williamdes,

Thanks for stepping up to assist. The package `libapache2-mod-php` indeed was **not** installed previously in the system. I did install it per your suggestion. Unfortunately, I am still unable to use **phpMyAdmin** as I used to in previous versions of Ubuntu (including 19.04!). Navigating to the installed phpmyadmin folder via `http://localhost/phpmyadmin/ `still results in the raw PHP code being display and not the accustomed db server logon/connection home page. Perhaps, you may want me to try some more configuration tests/checks? Much appreciated.

Kind regards.

P.S. I used the _temporary_/_custom_ **ppa** method to get the **phpMyAdmin** package for my 19.10 environment. I sincerely hope that I do not have to wait for Ubuntu 20.04 where the need for the temporary **ppa** has been made redundant.

**Comment by williamdes** [comment]
Hello @baqwas 

Please send me an email "williamdes at wdes.fr" or open a question on StackOverflow so you can have some help.

I would prefer not to add to much comments here so that other internet users do not have to scroll and load at lot of unrelated comments ;)

> I sincerely hope that I do not have to wait for Ubuntu 20.04 where the need for the temporary ppa has been made redundant.

I am sorry but as they said (Ubuntu team) https://github.com/phpmyadmin/phpmyadmin/issues/15515#issuecomment-559779931 they will not re-introduce the package in 19.10

Sorry for the late response.

**Comment by tawfiek** [comment]
> Here is the PPA rep link:
> 
> > **sudo add-apt-repository ppa:phpmyadmin/ppa**


E: The repository 'http://ppa.launchpad.net/jfi/ppa/ubuntu eoan Release' does not have a Release file

**Comment by williamdes** [comment]
@Blaimi could you please have a look?

**Comment by Blaimi** [comment]
> > Here is the PPA rep link:
> > > **sudo add-apt-repository ppa:phpmyadmin/ppa**
> 
> E: The repository 'http://ppa.launchpad.net/jfi/ppa/ubuntu eoan Release' does not have a Release file

The ppa with the error is the one from `jfi` for package `psensor` not for phpMyAdmin.

Execute `sudo add-apt-repository --remove ppa:jfi/ppa`and the error should be gone.

**Comment by williamdes** [comment]
Hi everyone
I published 4.9.4 to the [Ubuntu PPA](https://launchpad.net/~phpmyadmin/+archive/ubuntu/ppa) :rocket: !
Only available for Eoan and Bionic.

**Comment by vermajnv** [comment]
> @OGJAYYOU you need to do
> 
> ```shell
> sudo add-apt-repository ppa:phpmyadmin/ppa
> sudo apt-get update
> sudo apt-get install phpmyadmin
> ```

Thanks @williamdes ,
This works for me.

**Comment by baqwas** [comment]
v4.9.4 worked on my _Eoan_ platform. Thx again @williamdes for your persistent efforts. **You** and **phpmyadmin** keep me productive.

Kind regards.

**Comment by williamdes** [comment]
Hi !
I published 4.9.5 to the [Ubuntu PPA](https://launchpad.net/~phpmyadmin/+archive/ubuntu/ppa) :rocket:
- updated tcpdf to 6.3.5
- updated sql-parser to 4.6.1

Only available for Eoan and Bionic.

**Comment by baqwas** [comment]
Thank you for your continuing efforts.

Kind regards.

**Comment by williamdes** [comment]
I published motranslator 5 to the PPA for Eoan and Bionic because of https://github.com/phpmyadmin/phpmyadmin/issues/15236#issuecomment-604871109

Explained in https://github.com/phpmyadmin/phpmyadmin/issues/15236#issuecomment-604892735

Happy apt-get upgrade :rocket:
